### PR TITLE
SUP-21354: produce error log for unknown encoding upon any CSV bulk u…

### PR DESCRIPTION
**Pull request by avichainoach**
…pload

### CODE REVIEW

File: plugins/bulk_upload/csv/batch/BulkUploadEngineCsv.php [hunk 1]
```diff
@@ -157,27 +157,48 @@ protected function addBulkUploadResult(KalturaBulkUploadResult $bulkUploadResult
...
+
+		if ($foundUnknownEncoding)
+		{
+			$bulkUploadResult->objectStatus = KalturaUserStatus::ACTIVE;
+			$bulkUploadResult->status = KalturaBulkUploadResultStatus::ERROR;
+			$bulkUploadResult->errorType = KalturaBatchJobErrorTypes::APP;
+			$bulkUploadResult->errorDescription = 'value with unknown encoding was passed';
+
+			$this->addBulkUploadResult($bulkUploadResult);
+			return null;
```
Comment (MosheMaorKaltura) [inline plugins/bulk_upload/csv/batch/BulkUploadEngineCsv.php:199]: Should we return null or the defected results?

File: plugins/bulk_upload/csv/batch/BulkUploadEngineCsv.php [hunk 2]
```diff
@@ -157,27 +157,48 @@ protected function addBulkUploadResult(KalturaBulkUploadResult $bulkUploadResult
...
 			return null;
 		}
 		$this->handledRecordsThisRun++;
-		
-	    $bulkUploadResult = $this->getUploadResultInstance();
+
+		$foundUnknownEncoding = false;
+
+		$bulkUploadResult = $this->getUploadResultInstance();
+		$bulkUploadResult->bulkUploadResultObjectType = $this->getUploadResultInstanceType();
```
[the remark this answers is not in the document]
Comment (avichainoach) [reply plugins/bulk_upload/csv/batch/BulkUploadEngineCsv.php:170]: I did it, line 266

File: plugins/bulk_upload/csv/batch/BulkUploadEngineCsv.php (hunk 1 above)
Comment (avichainoach) [reply plugins/bulk_upload/csv/batch/BulkUploadEngineCsv.php:199]: null.
we're dealing with it one line before, by 'addBulkUploadResult'.
then returning null, same if we reached 'maxRecordsEachRun' at the beginning of this function.

**Comment by MosheMaorKaltura** [comment]
reviewed

**Pull request merged**